### PR TITLE
Feature/rm 9067 support for date time 2 sql data type

### PR DESCRIPTION
**Pull request by Frazerus**
https://re-motion.atlassian.net/browse/RM-9067

### CODE REVIEW

File: SecurityManager/Clients.Web.Test.SetupDB/SetupDB.sql [hunk 1]
```diff
@@ -165,12 +165,12 @@ CREATE TYPE [dbo].[TVP_Double_Distinct] AS TABLE
...
-CREATE TYPE [dbo].[TVP_DateTime] AS TABLE
+CREATE TYPE [dbo].[TVP_DateTime2] AS TABLE
 (
   [Value] datetime NULL
 )
 GO
-CREATE TYPE [dbo].[TVP_DateTime_Distinct] AS TABLE
+CREATE TYPE [dbo].[TVP_DateTime2_Distinct] AS TABLE
 (
   [Value] datetime NULL
```
Comment (a-ctor) [inline SecurityManager/Clients.Web.Test.SetupDB/SetupDB.sql:175]: Is it intentional that we did not change the column types to `datetime2`?

File: SecurityManager/Core/Database/SecurityManagerSetupDB.sql [hunk 2]
```diff
@@ -574,12 +574,12 @@ CREATE TYPE [dbo].[TVP_Double_Distinct] AS TABLE
...
-CREATE TYPE [dbo].[TVP_DateTime] AS TABLE
+CREATE TYPE [dbo].[TVP_DateTime2] AS TABLE
 (
   [Value] datetime NULL
 )
 GO
-CREATE TYPE [dbo].[TVP_DateTime_Distinct] AS TABLE
+CREATE TYPE [dbo].[TVP_DateTime2_Distinct] AS TABLE
 (
   [Value] datetime NULL
```
Comment (a-ctor) [inline SecurityManager/Core/Database/SecurityManagerSetupDB.sql:584]: Same question as above, why did we not change the columns?

**Review by a-ctor — APPROVED** [review]
The changes look good but I can't determine if all the required changes were made.

File: SecurityManager/Clients.Web.Test.SetupDB/SetupDB.sql (hunk 1 above)
Comment (Frazerus) [reply SecurityManager/Clients.Web.Test.SetupDB/SetupDB.sql:175]: Noep

**Review by MichaelKetting — APPROVED** [review]

**Pull request merged**